extension.js

```diff
@@ -125,24 +126,29 @@ var ContainerSubMenuMenuItem = GObject.registerClass(
     class extends PopupMenu.PopupSubMenuMenuItem {
         _init(container) {
             super._init(container.name);
-            this.menu.addMenuItem(new PopupMenuItem("Status", container.status));
-            this.menu.addMenuItem(new PopupMenuItem("Id", container.id));
-            this.menu.addMenuItem(new PopupMenuItem("Image", container.image));
-            this.menu.addMenuItem(new PopupMenuItem("Command", container.command));
-            this.menu.addMenuItem(new PopupMenuItem("Created", container.createdAt));
-            this.menu.addMenuItem(new PopupMenuItem("Ports", container.ports));
-            const ipAddrMenuItem = new PopupMenuItem("IP Address", "");
-            this.menu.addMenuItem(ipAddrMenuItem);
-            this.inspected = false;
-
-            // add more stats and info - inspect - SLOW
-            this.connect("button_press_event", () => {
-                if (!this.inspected) {
-                    container.inspect();
-                    this.inspected = true;
-                    ipAddrMenuItem.label.set_text(`${ipAddrMenuItem.label.text} ${container.ipAddress}`);
-                }
-            });
+            const actions = new PopupMenu.PopupBaseMenuItem({reactive: false, can_focus: false});
+            const details = new PopupMenu.PopupMenuSection();
+            this.menu.addMenuItem(actions);
+            this.menu.addMenuItem(new PopupMenu.PopupSeparatorMenuItem());
+            this.menu.addMenuItem(details);
+
+            const startBtn = addButton(actions, () => container.start(), "media-playback-start-symbolic");
+            const stopBtn = addButton(actions, () => container.stop(), "media-playback-stop-symbolic");
+            const restartBtn = addButton(actions, () => container.restart(), "system-reboot-symbolic");
+            const deleteBtn = addButton(actions, () => container.rm(), "user-trash-symbolic.symbolic");
+            const pauseBtn = addButton(
+                actions,
+                () => {
+                    if (container.status.split(" ")[0] === "running") {
+                        container.pause();
+                    }
+                    if (container.status.split(" ")[0] === "paused") {
+                        container.unpause();
+                    }
+                },
+                "media-playback-pause-symbolic"
+            );
+            pauseBtn.toggle_mode = true;
```

Comment:
I'd do
```js
    pauseBtn.checked = container.status.startsWith("paused");
```
here.

It doesn't look like you ever set "checked", and only reset it in the up/running case.